tools/src/golangci-lint/pin.go

```diff
@@ -1,6 +1,11 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
 //go:build pin
 // +build pin
 
 package ignore
 
-import "github.com/golangci/golangci-lint/cmd/golangci-lint"
+import _ "github.com/golangci/golangci-lint/cmd/golangci-lint"
```

Comment:
Adding these makes these imports not get removed by `gofmt`, so auto-format-on-save won't break them.

Reply:
`gofmt` won't remove them, but `goimports` (which is mostly a drop-in replacement for `gofmt`, and is what your auto-format-on-save is likely using) will.

Reply:
But this is also going to require a change to the regex in `tools.mk`

```patch
-	cd $(<D) && GOOS= GOARCH= go build -o $(abspath $@) $$(sed -En 's,^import "(.*)".*,\1,p' pin.go)
+	cd $(<D) && GOOS= GOARCH= go build -o $(abspath $@) $$(sed -En 's,^import _ "(.*)".*,\1,p' pin.go)
```

Reply:
Oh, that'll teach me to work on this late. Adding the underscore breaks the whole tooling, Go doesn't consider the import relevant any more and so can't do `go build`, it seems. Sigh, I'll roll it back, no editing `pin.go` without using a separate editor for me.